util/conditions/setter.go

```diff
@@ -42,31 +42,37 @@ func Set(to Setter, condition *clusterv1.Condition) {
 		return
 	}
 
+	// Check if the new conditions already exists, and change it only if there is a status
+	// transition (otherwise we should preserve the current last transition time)-
 	conditions := to.GetConditions()
-	newConditions := make(clusterv1.Conditions, 0, len(conditions))
 	exists := false
-	condition.LastTransitionTime = metav1.NewTime(time.Now().Truncate(time.Second))
 	for i := range conditions {
 		existingCondition := conditions[i]
 		if existingCondition.Type == condition.Type {
 			exists = true
 			if !hasSameState(&existingCondition, condition) {
-				newConditions = append(newConditions, *condition)
-				continue
+				condition.LastTransitionTime = metav1.NewTime(time.Now().UTC().Truncate(time.Second))
+				conditions[i] = *condition
 			}
+			condition.LastTransitionTime = existingCondition.LastTransitionTime
+			break
 		}
-		newConditions = append(newConditions, existingCondition)
 	}
+
+	// If the condition exists, adds it, setting the transition time only if not already set
```

Comment:
This comment doesn't seem to quite align to the logic
```suggestion
	// If the condition does not exist, add it, setting the transition time only if not already set
```